src/Materials/Textures/babylon.videoTexture.ts

```diff
@@ -1,44 +1,116 @@
 ﻿module BABYLON {
+    /**
+     * Settings for finer control over video usage
+     * @typedef {Object} VideoTextureSettings
+     * @property {boolean} [autoPlay] Applies `autoplay` to video, if specified
+     * @property {boolean} [loop] - Applies `loop` to video, if specified
+     * @property {boolean} autoUpdateTexture - Automatically updates internal texture from video at every frame in the render loop
+     */
+    export interface VideoTextureSettings {
+        autoPlay?: boolean;
+        loop?: boolean;
+        autoUpdateTexture: boolean;
+    }
+
+    const getName = (src: string | string[] | HTMLVideoElement): string => {
+        if (src instanceof HTMLVideoElement) {
+            return src.currentSrc;
+        }
+
+        if (typeof src === "object") {
+            return src.toString();
+        }
+
+        return src;
+    };
+
+    const getVideo = (src: string | string[] | HTMLVideoElement): HTMLVideoElement => {
+        if (src instanceof HTMLVideoElement) {
+            return src;
+        }
+        const video: HTMLVideoElement = document.createElement("video");
+        if (typeof src === "string") {
+            video.src = src;
+        } else {
+            src.forEach(url => {
+                const source = document.createElement("source");
+                source.src = url;
+                video.appendChild(source);
+            });
+        }
+        return video;
+    };
+
     export class VideoTexture extends Texture {
-        public video: HTMLVideoElement;
+        readonly autoUpdateTexture: boolean;

```

Comment:
this is public by default and thus it requires a comment (and please add public keyword)